services/secrets/secrets.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package secrets
+
+import (
+	"context"
+	"fmt"
+
+	auth_model "code.gitea.io/gitea/models/auth"
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/builder"
+)
+
+// MasterKeyProviderType is the type of master key provider
+type MasterKeyProviderType string
+
+// Types of master key providers
+const (
+	MasterKeyProviderTypeNone  MasterKeyProviderType = "none"
+	MasterKeyProviderTypePlain MasterKeyProviderType = "plain"
+)
+
+var (
+	masterKey   MasterKeyProvider
+	encProvider EncryptionProvider
+)
+
+// Init initializes master key provider based on settings
+func Init() error {
+	switch MasterKeyProviderType(setting.MasterKeyProvider) {
+	case MasterKeyProviderTypeNone:
+		masterKey = NewNopMasterKeyProvider()
+	case MasterKeyProviderTypePlain:
+		masterKey = NewPlainMasterKeyProvider()
+	default:
+		return fmt.Errorf("invalid master key provider %v", setting.MasterKeyProvider)
+	}
+
+	if err := masterKey.Init(); err != nil {
+		return err
+	}
+
+	encProvider = NewAesEncryptionProvider()
+
+	return nil
+}
+
+// GenerateMasterKey generates a new master key and returns secret or secrets for unsealing
+func GenerateMasterKey() ([][]byte, error) {
+	return masterKey.GenerateMasterKey()
+}
+
+func Encrypt(secret []byte) ([]byte, error) {
+	key, err := masterKey.GetMasterKey()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(key) == 0 {
+		return secret, nil
+	}
+
+	return encProvider.Encrypt(secret, key)
+}
+
+func EncryptString(secret string) (string, error) {
+	key, err := masterKey.GetMasterKey()
+	if err != nil {
+		return "", err
+	}
+
+	if len(key) == 0 {
+		return secret, nil
+	}
+
+	return encProvider.EncryptString(secret, key)
+}
+
+func Decrypt(enc []byte) ([]byte, error) {
+	key, err := masterKey.GetMasterKey()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(key) == 0 {
+		return enc, nil
+	}
+
+	return encProvider.Decrypt(enc, key)
+}
+
+func DecryptString(enc string) (string, error) {
+	key, err := masterKey.GetMasterKey()
+	if err != nil {
+		return "", err
+	}
+
+	if len(key) == 0 {
+		return enc, nil
+	}
+
+	return encProvider.DecryptString(enc, key)
+}
+
+func InsertRepoSecret(ctx context.Context, repoID int64, key, data string) error {
+	v, err := EncryptString(data)
+	if err != nil {
+		return err
+	}
+	return db.Insert(ctx, &auth_model.Secret{
+		RepoID: repoID,
+		Name:   key,
+		Data:   v,
+	})
+}
+
+func InsertOrgSecret(ctx context.Context, userID int64, key, data string) error {
```

Comment:
There is a name mismatch here: `InsertOrgSecret` and `FindUserSecrets`. I would rename both to `Owner`. Maybe change the structs member too.

Reply:
I agree about the struct fields.
But we should find a name to stick with as `OwnerSecret` sounds abolutely wrong to me and doesn't clarify the difference to `RepoSecret`s enough.
What about `UserSecret`s instead, even if they can also be for an organization?